inference-engine/src/mkldnn_plugin/mkldnn_memory.cpp

```diff
@@ -335,21 +325,27 @@ void MKLDNNMemory::redefineDesc(MemoryDescPtr desc) {
 }
 
 template<>
-BlockedMemoryDesc MKLDNNMemory::GetDescWithType<BlockedMemoryDesc, 0, 0>() const {
+CpuBlockedMemoryDesc MKLDNNMemory::GetDescWithType<CpuBlockedMemoryDesc, 0, 0>() const {
     if (auto descPtr = dynamic_cast<const BlockedMemoryDesc*>(pMemDesc.get())) {
-        return *descPtr;
+        return CpuBlockedMemoryDesc(descPtr->getPrecision(), descPtr->getShape(), descPtr->getBlockDims(),
+                                    descPtr->getOrder(), descPtr->getOffsetPadding(), descPtr->getOffsetPaddingToData(), descPtr->getStrides());
     } else {
-        switch (pMemDesc->getType()) {
-            case (MemoryDescType::Mkldnn):
-                return MemoryDescUtils::convertToBlockedDescriptor(*(pMemDesc->as<MKLDNNMemoryDesc>()));
-            default:
-                IE_THROW() << "Can not convert unsupported memory descriptor";
-        }
+        IE_THROW() << "Can not convert unsupported memory descriptor";
+    }
+}
+
+template<>
+OnednnBlockedMemoryDesc MKLDNNMemory::GetDescWithType<OnednnBlockedMemoryDesc, 0, 0>() const {
+    if (auto descPtr = dynamic_cast<const BlockedMemoryDesc*>(pMemDesc.get())) {
+        return OnednnBlockedMemoryDesc(descPtr->getPrecision(), descPtr->getShape(), descPtr->getBlockDims(),
+                                       descPtr->getOrder(), descPtr->getOffsetPadding(), descPtr->getOffsetPaddingToData(), descPtr->getStrides());
+    } else {
+        IE_THROW() << "Can not convert unsupported memory descriptor";
     }
 }
 
 bool MKLDNNMemoryDesc::operator==(const MKLDNNMemoryDesc &rhs) const {
-    return this->desc == rhs.desc && order == rhs.order;
+    return this->desc == rhs.desc;
```

Comment:
We need additional check whether the dynamic type is `OnednnBlocked` or not to execute the proper specialization.

Reply:
looks like we don't use them, so i suggest to remove them